File: apps/api/src/app/history/history.entity.ts
import { Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { ExercisesEntity } from '../exercises/exercises.entity';
import { WorkoutEntity } from '../workout/workout.entity';

@Entity({ name: 'history', schema: 'public' })
export class HistoryEntity {

    @PrimaryGeneratedColumn('increment')
    id: number;

    @ManyToOne(() => ExercisesEntity, (exercise) => exercise.history, { eager: true })
    @JoinColumn({ name: 'exercise_id' })
    exercise: ExercisesEntity;

    @ManyToOne(() => WorkoutEntity, workout => workout.history, { eager: true })
    @JoinColumn({ name: 'workout_id' })
    workout: WorkoutEntity;

    @Column({ type: 'float' })
    weight: number;

    @Column({ name: 'created_at' })
    createdAt: Date;
}

File: my-workout-api/src/main/java/com/avernet/my_workout_api/History/HistoryEntity.java
package com.avernet.my_workout_api.History;

import com.avernet.my_workout_api.Workout.WorkoutEntity;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "muscle_group")
@Data
public class HistoryEntity {
    @Id
    Long id;

    @Column()
    Float weight;

    @Column()
    Short reps;

    @ManyToOne(fetch = FetchType.LAZY)
    WorkoutEntity workout;
}
